### PR TITLE
EVG-8113 fix project setup commands

### PR DESCRIPTION
**Pull request by ybrill**
The working directory for commands is validated before we actually run the command so we need to mkdir first.
When the directory doesn't exist we prepend a mkdir command. Subsequent commands will have their working directory set by the PreHook.

A separate bug was the command logging. When we log the command being executed we were logging the last project command for every command. The reason appears to be that when the Prerequisite function executes it captures the current value of the obj variable (which changes as we iterate over the slice). Storing in a separate variable fixes the issue.

### CODE REVIEW

**Comment by ybrill** [comment]
People run it manually. Theoretically, someone with an old workstation could run it.

File: model/project_ref.go [hunk 1]
```diff
@@ -957,15 +958,20 @@ func (p *ProjectRef) GetProjectSetupCommands(opts apimodels.WorkstationSetupComm
...
+		}
 
-		commandNumber := idx + 1 // to avoid logging a stale number
-		cmd := jasper.NewCommand().Directory(dir).AppendArgs("mkdir", "-p", dir).
-			SetErrorSender(level.Error, opts.Output).
+		// avoid logging the final value of obj
+		commandNumber := idx + 1
+		cmdString := obj.Command
+
+		cmd := jasper.NewCommand().SetErrorSender(level.Error, opts.Output).
```
Comment (ablack12) [inline model/project_ref.go]: Do we still need to set Directory(dir)?

File: model/project_ref.go [hunk 2]
```diff
@@ -957,15 +958,20 @@ func (p *ProjectRef) GetProjectSetupCommands(opts apimodels.WorkstationSetupComm
...
 			Append(obj.Command).
```
Comment (ablack12) [inline model/project_ref.go:966]: can we use cmdString here now as well?

File: model/project_ref.go (hunk 1 above)
Comment (ybrill) [reply model/project_ref.go]: I think so (otherwise the command will run in the user's home directory, I think)

Comment (ybrill) [reply model/project_ref.go]: Ah, I get it. The latest commit added it back

File: model/project_ref.go (hunk 2 above)
Comment (ybrill) [reply model/project_ref.go:966]: we could, but they should both be the same

**Pull request merged**